indico/modules/events/papers/schemas.py

```diff
@@ -0,0 +1,122 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2019 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+from markupsafe import escape
+from marshmallow.fields import Boolean, DateTime, Field, Float, Function, Integer, List, Nested, String
+from marshmallow_enum import EnumField
+
+from indico.core.marshmallow import mm
+from indico.modules.events.contributions.schemas import ContributionSchema
+from indico.modules.events.models.events import Event
+from indico.modules.events.papers.models.comments import PaperReviewComment
+from indico.modules.events.papers.models.files import PaperFile
+from indico.modules.events.papers.models.reviews import PaperReview
+from indico.modules.events.papers.models.revisions import PaperRevision, PaperRevisionState
+from indico.modules.users.schemas import UserSchema
+from indico.util.mimetypes import icon_from_mimetype
+
+
+class PaperEventSchema(mm.ModelSchema):
+    class Meta:
+        model = Event
+        fields = ('id', 'title', 'is_locked')
+
+
+class PaperFileSchema(mm.ModelSchema):
+    icon = Function(lambda paper_file: icon_from_mimetype(paper_file.content_type))
+
+    class Meta:
+        model = PaperFile
+        fields = ('id', 'revision_id', 'content_type', 'filename', 'icon')
+
+
+class PaperRevisionStateSchema(mm.Schema):
+    name = String()
+    title = String()
+    css_class = String()
+
+
+class PaperAction(mm.Schema):
+    name = String()
+    css_class = String()
+
+
+class PaperCommentVisibilitySchema(mm.Schema):
+    name = String()
+    title = String()
+
+
+class PaperRevisionTimelineField(Field):
+    def _serialize(self, value, attr, obj, **kwargs):
+        if not value:
+            return []
+
+        serialized = []
+        for timeline_item in value:
+            serialized_item = {'timeline_item_type': timeline_item.timeline_item_type}
+            if timeline_item.timeline_item_type == 'comment':
+                serialized_item.update(paper_review_comment_schema.dump(timeline_item))
+            elif timeline_item.timeline_item_type == 'review':
+                serialized_item.update(paper_review_schema.dump(timeline_item))
+            serialized.append(serialized_item)
+        return serialized
+
+
+class PaperRevisionSchema(mm.ModelSchema):
+    submitter = Nested(UserSchema)
+    judge = Nested(UserSchema)
+    spotlight_file = Nested(PaperFileSchema)
+    files = List(Nested(PaperFileSchema))
+    state = EnumField(PaperRevisionState)
+    timeline = PaperRevisionTimelineField()
+    judgment_comment = Function(lambda revision: escape(revision.judgment_comment))
```

Comment:
IIRC we said that we wanted to have `judgment_comment_html` etc. for the rendered versions. This will be more important for anything that's editable, since you need the source version there as well.